src/main/cpp/auton/drivePrimitives/DrivePathPlanner.cpp

```diff
@@ -187,47 +230,41 @@ bool DrivePathPlanner::IsDone()
+
+        if (m_zone->GetZoneMode() == AutonGrid::ZoneMode::CIRCLE)
         {
-            Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "Distance To Reef Branch", "Reef Branch Found: ", true);
-            auto branchPose = get<1>(reefBranch.value());
-
-            // check if we see a Reef Branch is one we want to get
-
-            Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "Distance To Reef Branch", "Consider Reef Branch: ", true);
-
-            auto chassispose = m_chassis->GetPose();
-            auto distanceToBranch = chassispose.Translation().Distance(branchPose.Translation());
-
-            auto currentTime = m_timer.get()->Get();
-            auto percent = currentTime.value() / m_totalTrajectoryTime.value();
-            Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "Distance To Reef Branch", "time:", currentTime.value());
-            Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "Distance To Reef Branch", "Done Percent:", static_cast<double>((currentTime.value()) / m_totalTrajectoryTime.value()));
-            Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "Distance To Reef Branch", "Distance: ", distanceToBranch.value());
-
-            if (distanceToBranch <= m_distanceThreshold) // switch to drive to Reef Branch
-            {
-                Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "Distance To Branch", "Switch to Drive To Reef Branch: ", true);
-
-                m_pathname = "RIGHT_REEF_BRANCH";
-                m_isVisionDrive = true;
-                m_visionAlignment = PrimitiveParams::VISION_ALIGNMENT::REEF;
-                InitMoveInfo();
-            }
-            else
-            {
-                Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "Distance To Reef Branch", "Switch to Drive To Reef Branch: ", false);
-            }
+            isInZone = AutonGrid::GetInstance()->IsPoseInZone(m_zone->getCircleZonePose(),
+                                                              m_zone->getRadius(),
+                                                              m_chassis->GetPose());
         }
-        else
+        else if (m_zone->GetZoneMode() == AutonGrid::ZoneMode::RECTANGLE)
         {
-            Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "Distance To Reef Branch", "Reef Branch Found: ", false);
+            isInZone = AutonGrid::GetInstance()->IsPoseInZone(m_zone->GetXGrid1(),
+                                                              m_zone->GetXGrid2(),
+                                                              m_zone->GetYGrid1(),
+                                                              m_zone->GetYGrid2(),
+                                                              m_chassis->GetPose());
         }
     }
+
+    if (isInZone) // switch to drive to Reef Branch
+    {
+        Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "Distance To Branch", "Switch to Drive To Reef Branch: ", true);
+
```

Comment:
There are probably a bunch of things to come back to as many things refer to reef branches or right reef branch.  This might be able to be a generic method that handles all of the driveto options